src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -69,9 +70,24 @@ public void Dispose()
         protected virtual void Dispose(bool disposing)
         {
             // No managed resources to clean up, so disposing is ignored.
+            try
+            {
+                try
+                {
+                    Directory.Delete(TestDirectory, recursive: true);
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    DirectoryInfo di = new DirectoryInfo(TestDirectory);
+                    foreach (FileInfo file in di.GetFiles("*", SearchOption.AllDirectories))
```

Comment:
Just noticed this should be FileSystemInfo -- it needs to handle directories also